nupic/research/frameworks/pytorch/imagenet/imagenet_experiment.py

```diff
@@ -719,7 +719,12 @@ def get_state(self):
 
         if self.lr_scheduler is not None:
             with io.BytesIO() as buffer:
-                serialize_state_dict(buffer, self.lr_scheduler.state_dict())
+                state_dict = self.lr_scheduler.state_dict()
+                if "anneal_func" in state_dict:
+                    # FIXME: This is a workaround for a PyTorch bug.
+                    # https://github.com/pytorch/pytorch/issues/42376
+                    del state_dict["anneal_func"]
```

Comment:
You should store the `anneal_strategy` instead and restore the correct function on load.

Reply:
I think that would be overkill. The state dict is used for time-varying attributes. There is no supported way to change the anneal strategy after the LR scheduler is instantiated. They include it in the state dict because they're lazily saving the whole `self.__dict__` rather than writing proper `state_dict` methods.